common/rest/middleware/changeLog.py

```diff
@@ -0,0 +1,45 @@
+import uuid
+import socket
+from contextvars import ContextVar
+
+_request_ctx = ContextVar('current_request', default={})
+
+def set_current_request(new_context):
+    _request_ctx.set(new_context)
+
+def get_current_request():
+    return _request_ctx.get()
+
+def add_to_request_context(key, value):
+    context = _request_ctx.get()
+    context[key] = value
+    _request_ctx.set(context)
+
+def get_ip_address():
+    try:
```

Comment:
This method should get ip public on the server not from x-real or x-forwarded-for this is not true logic.

Reply:
Fixed. The method now correctly retrieves the public IP from the server.